src/flamenco/runtime/tests/fd_exec_instr_test.c

```diff
@@ -515,11 +545,363 @@ _context_create( fd_exec_instr_test_runner_t *        runner,
   return 1;
 }
 
+void
+_add_to_data(uchar ** data, void const * to_add, ulong size) {
+  while( size-- ) {
+    **data = *(uchar *)to_add;
+    (*data)++;
+    to_add = (uchar *)to_add + 1;
+  }
+}
+
+void
+_add_compact_u16(uchar ** data, ushort to_add) {
+  fd_bincode_encode_ctx_t encode_ctx = { .data = *data, .dataend = *data + 3 };  // Up to 3 bytes
+  fd_bincode_compact_u16_encode( &to_add, &encode_ctx );
+  *data = (uchar *) encode_ctx.data;
+}
+
+static int
+_txn_context_create( fd_exec_instr_test_runner_t *      runner,
+                     fd_exec_txn_ctx_t *                txn_ctx,
+                     fd_exec_slot_ctx_t *               slot_ctx,
+                     fd_exec_test_txn_context_t const * test_ctx ) {
+  fd_funk_t * funk = runner->funk;
+
+  /* Generate unique ID for funk txn */
+  static FD_TL ulong xid_seq = 0UL;
+
+  fd_funk_txn_xid_t xid[1] = {0};
+  xid->ul[0] = fd_log_app_id();
+  xid->ul[1] = fd_log_thread_id();
+  xid->ul[2] = xid_seq++;
+  xid->ul[3] = (ulong)fd_tickcount();
+
+  /* Create temporary funk transaction and scratch contexts */
+
+  fd_funk_txn_t * funk_txn = fd_funk_txn_prepare( funk, NULL, xid, 1 );
+  fd_scratch_push();
```

Comment:
Can we use the FD_SCRATCH_SCOPE_BEGIN/END macros

Reply:
Does not work because scratch gets popped whenever context is destroyed, which can happen in various locations

Reply:
You can create a scratch scope using the macros at the start of `fd_exec_txn_test_run`, and then end it at the end of `fd_exec_txn_test_run`. Then make it so that the context destroying doesn't manually pop. This might be cleaner.